pkg/platform/packet/template.go

```diff
@@ -192,22 +192,32 @@ module "worker-{{ $pool.Name }}" {
 }
 {{ end }}
 
-{{- if .Config.DNS.Provider.Manual }}
+{{- if eq .Config.DNSProvider "manual" }}
+module "dns" {
+  source = "../lokomotive-kubernetes/dns/manual"
+
+  cluster_name             = "{{ .Config.ClusterName }}"
+  controllers_public_ipv4  = module.packet-{{.Config.ClusterName}}.controllers_public_ipv4
+  controllers_private_ipv4 = module.packet-{{.Config.ClusterName}}.controllers_private_ipv4
+  dns_zone                 = "{{ .Config.DNSZone }}"
+}
+
 output "dns_entries" {
-  value = module.packet-{{.Config.ClusterName}}.dns_entries
+  value = module.dns.entries
 }
-{{- end }}
 
-{{- if .Config.DNS.Provider.Route53 }}
+{{- else if eq .Config.DNSProvider "route53" }}
```

Comment:
I've used `else` to make it clear that only one of the 3 DNS blocks can be rendered (since we don't have a "return" inside a template). If you think it's more readable to have 3 `end`s instead of 2 `else`s I'm fine with changing it. IMO the readability is about the same, so I'd rather have the `else`s as they clarify the intention better.

Reply:
With `else` you possibly need to look up what are the other statements to understand the flow of the template. With just 3 ends, you have fixed complexity of single no matter how many different providers you have, assuming the logic is always based on the same property.

Another thought which came to my mind, maybe we could just do `source = "../lokomotive-kubernetes/dns/{{ .Config.DNSProvider }}"`, as the parameters for the modules are always the same? This removes a lot of repetition, even if we need to have if blocks again for providers.

Reply:
Yes, your 2nd suggestion was where I was going with this initially. That's why I've converted the manual DNS option to a module and unified the vars. I thought it would be too messy because of the different `provider` blocks but if we don't need them at all I can give it a try.

Reply:
Indeed, we can get rid of the `provider` Terraform blocks in the template. The `dns` module is generic now.